build.gradle

```diff
@@ -40,9 +40,8 @@ subprojects { subProject ->
     tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
         kotlinOptions {
             jvmTarget = "1.8"
-            apiVersion = "1.4"
-            languageVersion = "1.4"
-            useIR = findProperty("kotlin.ir.enabled")?.toBoolean() == true
+            apiVersion = "1.7"
+            languageVersion = "1.7"
```

Comment:
And in fact, builds with lower kotlin versions fail.

Reply:
Yes, I'm not sure what's best (I tried to explain my thoughts in the PR description). It would break the library for users not on 1.7, but only for new versions - they can still use the current version of mockk.

To properly support value classes (the reason I'm making this PR, see #849) I think it should at least be set to 1.5, so the [`isValue`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/-k-class/is-value.html) flag is available. This means dropping support for 1.4. I think 1.4 is deprecated, at least that's what Gradle tells me every time I run a build! But I can't find any documentation supporting this.

Reply:
Also, can you please update the github actions yml file to remove support for 1.4 and add 1.7?

Reply:
okay, done!